worlds/jakanddaxter/JakAndDaxterOptions.py

```diff
@@ -0,0 +1,141 @@
+    you will add 8 items to the pool."""
+    display_name = "Enable Orbsanity"
+    option_off = 0
+    option_per_level = 1
+    option_global = 2
+    default = 0
+
+
+class GlobalOrbsanityBundleSize(Choice):
+    """Set the orb bundle size for Global Orbsanity. This only applies if "Enable Orbsanity" is set to "Global."
+    There are 2000 orbs in the game, so your bundle size must be a factor of 2000."""
+    display_name = "Global Orbsanity Bundle Size"
+    option_1_orb = 1
+    option_2_orbs = 2
+    option_4_orbs = 4
+    option_5_orbs = 5
+    option_8_orbs = 8
+    option_10_orbs = 10
+    option_16_orbs = 16
+    option_20_orbs = 20
+    option_25_orbs = 25
+    option_40_orbs = 40
+    option_50_orbs = 50
+    option_80_orbs = 80
+    option_100_orbs = 100
+    option_125_orbs = 125
+    option_200_orbs = 200
+    option_250_orbs = 250
+    option_400_orbs = 400
+    option_500_orbs = 500
+    option_1000_orbs = 1000
+    option_2000_orbs = 2000
+    default = 20
+
+
+class PerLevelOrbsanityBundleSize(Choice):
+    """Set the orb bundle size for Per Level Orbsanity. This only applies if "Enable Orbsanity" is set to "Per Level."
+    There are 50, 150, or 200 orbs per level, so your bundle size must be a factor of 50."""
+    display_name = "Per Level Orbsanity Bundle Size"
+    option_1_orb = 1
+    option_2_orbs = 2
+    option_5_orbs = 5
+    option_10_orbs = 10
+    option_25_orbs = 25
+    option_50_orbs = 50
+    default = 25
+
+
+class FireCanyonCellCount(Range):
+    """Set the number of power cells you need to cross Fire Canyon."""
+    display_name = "Fire Canyon Cell Count"
+    range_start = 0
+    range_end = 100
+    default = 20
+
+
+class MountainPassCellCount(Range):
+    """Set the number of power cells you need to reach Klaww and cross Mountain Pass."""
+    display_name = "Mountain Pass Cell Count"
+    range_start = 0
+    range_end = 100
+    default = 45
+
+
+class LavaTubeCellCount(Range):
+    """Set the number of power cells you need to cross Lava Tube."""
+    display_name = "Lava Tube Cell Count"
+    range_start = 0
+    range_end = 100
+    default = 72
+
+
+# 222 is the maximum because there are 9 citizen trades and 2000 orbs to trade (2000/9 = 222).
+class CitizenOrbTradeAmount(Range):
+    """Set the number of orbs you need to trade to ordinary citizens for a power cell (Mayor, Uncle, etc.).
+
+    Along with Oracle Orb Trade Amount, this setting cannot exceed the total number of orbs in the game (2000)."""
+    display_name = "Citizen Orb Trade Amount"
+    range_start = 0
+    range_end = 222
+    default = 90
+
+
+# 333 is the maximum because there are 6 oracle trades and 2000 orbs to trade (2000/6 = 333).
+class OracleOrbTradeAmount(Range):
+    """Set the number of orbs you need to trade to the Oracles for a power cell.
+
+    Along with Citizen Orb Trade Amount, this setting cannot exceed the total number of orbs in the game (2000)."""
```

Comment:
That would be [here](https://github.com/ArchipelagoMW/Archipelago/blob/b7ca9cbc2f567278fd9abb9ccedd7231e0bddbee/worlds/jakanddaxter/Regions.py#L190) but I agree we can provide the equation in the comment just to cover our bases.

Reply:
lol, it's in the one file I hadn't looked at, of course XD
Ideally, you'd want errors from that to happen as soon as possible, like in `generate_early`, I haven't checked yet if those errors actually rely on `create_regions` and whether you can only determine whether or not they'll error after that, but if they _can_ be moved earlier, they should.

As far as `assert` vs not having an assert goes, if the error is something you suspect could never happen and that would not have significant or irreparable impact on a multiworld, an assert would be good because it wouldn't run (use up time) in the frozen/released builds of AP. But if you expect it will get hit, and it causes problems that, say, require restarting the multiworld or using server commands like `/send` and `/send_location` to fix, it should throw an actual error.